tmt/cli.py

```diff
@@ -2030,20 +2028,30 @@ def setup_completion(shell: str, install: bool) -> None:
     else:
         script = Path(config.path) / f'{COMPLETE_SCRIPT}.{shell}'
 
-    # SIM115: Use context handler for opening files. Would not reduce complexity here.
-    out = open(script, 'w') if install else sys.stdout  # noqa: SIM115
-    subprocess.run(f'{COMPLETE_VARIABLE}={shell}_source tmt',
-                   shell=True, stdout=out)
+    env_var = {COMPLETE_VARIABLE: f'{shell}_source'}
+
+    logger = context.obj.logger
+
+    completions = Command('tmt').run(env=tmt.utils.Environment.from_dict(env_var),
+                                     cwd=None,
+                                     logger=context.obj.logger
+                                     ).stdout
```

Comment:
An issue for a future PR, do we really need to call `tmt` recursively? Feels weird to have a nested logger being created here.

Reply:
No comments from my side.